x/oracle/keeper/msg_server.go

```diff
@@ -228,17 +228,40 @@ func (k msgServer) DeleteTask(goCtx context.Context, msg *types.MsgDeleteTask) (
 func (k msgServer) CreateTxTask(goCtx context.Context, msg *types.MsgCreateTxTask) (*types.MsgCreateTxTaskResponse, error) {
 	ctx := sdk.UnwrapSDKContext(goCtx)
 
-	creatorAddr, err := sdk.AccAddressFromBech32(msg.Creator)
-	if err != nil {
-		return nil, err
+	creatorAddr, _ := sdk.AccAddressFromBech32(msg.Creator)
+	if msg.ValidTime.Before(ctx.BlockTime()) {
+		return nil, types.ErrOverdueValidTime
 	}
 
 	hashByte := sha256.Sum256(msg.TxBytes)
 	hash := base64.StdEncoding.EncodeToString(hashByte[:])
 
-	expirationTime := ctx.BlockTime().Add(k.Keeper.GetTaskParams(ctx).ExpirationDuration)
-	txTask := types.NewTxTask(msg.Creator, hashByte[:], msg.Bounty, msg.ValidTime, expirationTime, types.TaskStatusPending, nil)
-	if err := k.Keeper.CreateTask(ctx, creatorAddr, &txTask); err != nil {
+	var txTask *types.TxTask
+	// if a TaskStatusNil task already exists, overwrite it after copying several fields.
+	// please be noted that the expiration hook remains
+	if savedTask, err := k.GetTask(ctx, hashByte[:]); err == nil {
```

Comment:
is it better to move this part code to keeper.createtask?

Reply:
I agree. msgServer should mostly only route relevant fields into keeper functionalities

Reply:
it makes sense. 
done.